packages/flutter_tools/lib/src/windows/windows_device.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../application_package.dart';
+import '../base/platform.dart';
+import '../build_info.dart';
+import '../desktop.dart';
+import '../device.dart';
+import 'windows_workflow.dart';
+
+/// A device which represents a desktop Windows target.
+class WindowsDevice extends Device {
+  WindowsDevice() : super('windows_device');
+
+  @override
+  void clearLogs() {}
+
+  @override
+  DeviceLogReader getLogReader({ApplicationPackage app}) => NoOpDeviceLogReader('windows');
+
+  @override
+  Future<bool> installApp(ApplicationPackage app) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<bool> isAppInstalled(ApplicationPackage app) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<bool> isLatestBuildInstalled(ApplicationPackage app) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<bool> get isLocalEmulator async => false;
+
+  @override
+  bool isSupported() => true;
+
+  @override
+  String get name => 'windows';
```

Comment:
Nit: Since the strings in the iOS and Android device-related implementations are capitalized correctly, I'm assuming at least some of them are user-visible? If that's the case, it should be `'Windows'` (and similarly Linux and macOS)

Reply:
Strings are fixed and tested. Example MacOS:

```
MacOS • MacOS • darwin-x64 • Mac OS X 10.13.6
```

Its a little redundant because "device id" doesn't really make sense for a desktop target. Will clean up that logic later